File: src/main/java/io/github/tanguygab/conditionalactions/conditions/types/GroupCondition.java
package io.github.tanguygab.conditionalactions.conditions.types;

import io.github.tanguygab.conditionalactions.conditions.ConditionGroup;
import org.bukkit.OfflinePlayer;

import java.util.Map;

public class GroupCondition extends ConditionType {

    private final ConditionGroup group;

    public GroupCondition(ConditionGroup group, String... array) {
        super(array);
        this.group = group;
    }

    @Override
    public boolean isMet(OfflinePlayer player, Map<String, String> replacements) {
        return group.isMet(player,replacements);
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/conditions/types/GroupCondition.kt
package io.github.tanguygab.conditionalactions.conditions.types

import io.github.tanguygab.conditionalactions.conditions.ConditionGroup
import org.bukkit.OfflinePlayer

class GroupCondition(private val group: ConditionGroup, input: String) : ConditionType(listOf(input, "")) {
    override fun isMet(player: OfflinePlayer?, replacements: Map<String, String>) = group.isMet(player, replacements)
}
